### ISSUE TITLE
Don't contract nodes connecting edges with different names? Why?

### ISSUE THREAD

**Issue by nilsnolde** [valhalla/valhalla#4599]
I was debugging our TomTom converter and did a deeper analysis of the expansion where I found that also in OSM we break shortcuts fairly often when the names of two edges aren't exactly the same. For example here with `secondary` roads:
https://www.openstreetmap.org/way/294613493: has `ref=227` and `name=Žatecká`
https://www.openstreetmap.org/way/523627234: has `ref=227`

I can't see how names are important for expansion/costing, which is the only place where shortcuts matter right? Or is there some stuff in map-matching I'm not aware of? If so, would it be enough to have _some_ name match of the 2 edges (would resolve the above case)?

**Comment by nilsnolde** [comment]
Or they're used for narration too? Somehow I always figured we'd need to recover the base edges for that..

**Comment by nilsnolde** [comment]
Also why do the `speeds()` have to match for edges in the same direction? The shortcut gets a new speed anyways, based on the total length and duration, and it's quite common for lower class roads like tertiary to have alternating speeds.

**Comment by kevinkreiser** [comment]
yep you are right its a product of hysterical raisins! in the early days we didnt recover shortcuts before narrating the path i added that once we started supporting live traffic. anyway because we shipped paths with shortcuts to narrative/maneuver generation we couldnt contract on properties that are important to that process. now that we bit the bullet and implemented recovering the shortcuts we could drop this as a requirement for contracting a node.

basically its as you said, when making shortcuts back in the day we had to worry about both costing and narrative/maneuvers but now that we recover the shortcuts before the latter happens we only need to worry about costing during shortcut creation. good catch!
